### PR TITLE
Handle properly indexing rectangles that crosses the dateline

### PR DESCRIPTION
**Pull request by iverase**
When indexing a rectangle that crosses the dateline, we are currently not handling it properly and we index a polygon that do not cross the dateline. This changes generates two polygons wrapping the dateline.

### CODE REVIEW

**Comment by iverase** [comment]
@elasticmachine test this please

File: server/src/main/java/org/elasticsearch/index/mapper/GeoShapeIndexer.java [hunk 1]
```diff
@@ -261,7 +262,15 @@ public Void visit(Polygon polygon) {
 
         @Override
         public Void visit(Rectangle r) {
-            addFields(LatLonShape.createIndexableFields(name, GeoShapeUtils.toLucenePolygon(r)));
+            if (r.getMinLon() > r.getMaxLon()) {
+                Rectangle left = new Rectangle(r.getMaxX(), GeoUtils.MAX_LON, r.getMaxY(), r.getMinY());
+                addFields(LatLonShape.createIndexableFields(name, GeoShapeUtils.toLucenePolygon(left)));
+                Rectangle right = new Rectangle(GeoUtils.MIN_LON, r.getMaxLon(), r.getMaxY(), r.getMinY());
```
Comment (imotov) [inline server/src/main/java/org/elasticsearch/index/mapper/GeoShapeIndexer.java]: Maybe be better to stick with X, Y or Lat, Lon for clarity.

File: server/src/test/java/org/elasticsearch/search/geo/GeoShapeQueryTests.java [hunk 2]
```diff
@@ -791,4 +791,24 @@ public void testDistanceQuery() throws Exception {
             .get();
         assertEquals(0, response.getHits().getTotalHits().value);
     }
+
+    public void testIndexRectangleSpanningDateLine() throws Exception {
```
Comment (imotov) [inline server/src/test/java/org/elasticsearch/search/geo/GeoShapeQueryTests.java]: Could you add a unit test to GeometryIndexerTests for that use case as well?

**Review by imotov — APPROVED** [review]
Nice catch! Added a couple of minor requests. Otherwise LGTM.

File: server/src/main/java/org/elasticsearch/index/mapper/GeoShapeIndexer.java (hunk 1 above)
Comment (iverase) [reply server/src/main/java/org/elasticsearch/index/mapper/GeoShapeIndexer.java]: True! I will go for Lat / Lon

**Pull request merged**